File: scaffold-cli/templates/rust/src/routes/health.rs
//! Health check endpoint
//!
//! Provides a simple health check endpoint for load balancers and monitoring.

use axum::{
    extract::State,
    http::StatusCode,
    response::IntoResponse,
    Json,
};
use std::sync::Arc;

use crate::config::Config;
use crate::constants;
use crate::models::response::HealthResponse;

/// Health check handler
///
/// Returns a JSON response indicating service health.
pub async fn handler(
    State(config): State<Arc<Config>>,
) -> impl IntoResponse {
    let response = HealthResponse::healthy_with_version(
        constants::SERVICE_NAME,
        constants::VERSION,
    );

    (StatusCode::OK, Json(response))
}

#[cfg(test)]
mod tests {
    use super::*;

    #[tokio::test]
    async fn test_health_returns_200() {
        let config = Arc::new(Config::default());
        let response = handler(State(config)).await.into_response();
        assert_eq!(response.status(), StatusCode::OK);
    }
}

File: genlang-cli/templates/python/src/routes/health.py
"""
Health Check Endpoint

Provides a simple health check endpoint for load balancers and monitoring.
"""

from fastapi import APIRouter
from pydantic import BaseModel

from .. import __service__, __version__


router = APIRouter()


class HealthResponse(BaseModel):
    """Health check response model."""
    status: str
    service: str
    version: str | None = None


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """
    Health check endpoint.

    Returns a JSON response indicating service health.
    """
    return HealthResponse(
        status="healthy",
        service=__service__,
        version=__version__,
    )
